src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationMiddleware.cs

```diff
@@ -46,9 +45,14 @@ public class OpenIdConnectAuthenticationMiddleware : AuthenticationMiddleware<Op
         {
             _logger = loggerFactory.CreateLogger<OpenIdConnectAuthenticationMiddleware>();
 
+            if (string.IsNullOrEmpty(Options.SignInScheme))
+            {
+                Options.SignInScheme = externalOptions.Options.SignInScheme;
+            }
```

Comment:
Can we consider sharing the same options instead of introducing a similar flag?  i.e. OIDC can use ExternalAuthenticationOptions instead of looking at its similarly named (and used flag).  Hopefully this will make it clear that its used for exactly the same purpose as the other one...